FileSystem/FileIO.js

```diff
@@ -0,0 +1,254 @@
+/**
+ * File open mode enumeration
+ */
+var OpenMode = {
+  APPEND: 'append',
+  BINARY: 'binary',
+  INPUT: 'input',
+  OUTPUT: 'output',
+  RANDOM: 'random',
+};
+
+/**
+ * File lock mode enumeration
+ */
+var LockMode = {
+  SHARED: 'shared',
+  READ: 'read',
+  WRITE: 'write',
+  READ_WRITE: 'read_write',
+  NONE: 'none',
+};
+
+/**
+ * File access mode enumeration
+ */
+var AccessMode = {
+  READ: 'read',
+  WRITE: 'write',
+  READ_WRITE: 'read_write',
+};
+
+/**
+ * File Input Output library
+ */
+var FileIO = {
+  openFile: openFile,
+  closeFileList: closeFileList,
+  getNextAvailableFile: getNextAvailableFile,
+  lof: lof,
+  isEOF: isEOF,
+  printToFile: printToFile,
+
+  openFiles: {},
+
+  closeFile: closeFile,
+};
+
+/**
+ * Emulates VBA open statement API
+ * @todo Implement file locking
+ * @body File locking is not possible in Google Drive but it could be emulated by using PropertyServices.
+ * @param {string} path The local file path
+ * @param {number} fileNumber File number
+ * @param {string} openMode Open Mode enumeration
+ * @param {string} accessMode Access Mode enumeration
+ * @param {string} lockMode lockMode enumeration
+ */
+function openFile(path, fileNumber, openMode, accessMode, lockMode) {
+  // Set default arguments
+  openMode = openMode || OpenMode.RANDOM;
+  accessMode = accessMode || AccessMode.READ_WRITE;
+  lockMode = lockMode || LockMode.NONE;
+
+  // Check if file number is available
+  if (fileNumber in this.openFiles) {
+    throw Error('File Number: ' + fileNumber + ' is already being used');
+  }
+
+  // Check if file number is valid
+  if (fileNumber < 1 || fileNumber > 511) {
+    throw Error(
+      'File Number: ' +
+        fileNumber +
+        ' is invalid. File numbers need to be an number value between 1 and 511'
+    );
+  }
+
+  var fileId;
+
+  // If file exists, get file id else create and get file id
+  if (FileMapper.hasMapping(FileSystem.currentDirectory, path)) {
+    fileId = FileMapper.getFileId(FileSystem.currentDirectory, path);
+  } else {
+    fileId = createFile(FileSystem.currentDirectory, path, MimeType.PLAIN_TEXT);
+  }
+
+  // In memory file object
+  var file = {
+    fileId: fileId,
+    filePath: path,
+    accessMode: accessMode,
+    openMode: openMode,
+    lockMode: lockMode,
+  };
+
+  // Set file content (In memory buffer)
+  switch (openMode) {
```

Comment:
Update file.content and file.pointer in one switch block?

Reply:
The cases are slightly different, The Switch blocks are separate for readability. Would you still prefer them in 1 block?

Reply:
Yes, add comments in the code accordingly.

Reply:
Fixed